src/components/Modal/BaseModal.js

```diff
@@ -23,166 +24,182 @@ const defaultProps = {
     forwardedRef: () => {},
 };
 
-class BaseModal extends PureComponent {
-    constructor(props) {
-        super(props);
+function BaseModal({
+    isVisible,
+    onClose,
+    shouldSetModalVisibility,
+    onModalHide,
+    type,
+    popoverAnchorPosition,
+    innerContainerStyle,
+    outerStyle,
+    onModalShow,
+    propagateSwipe,
+    fullscreen,
+    animationIn,
+    animationOut,
+    useNativeDriver,
+    hideModalContentWhileAnimating,
+    animationInTiming,
+    animationOutTiming,
+    statusBarTranslucent,
+    onLayout,
+    avoidKeyboard,
+    forwardedRef,
+    children,
+}) {
+    const {windowWidth, windowHeight, isSmallScreenWidth} = useWindowDimensions();
+
+    const safeAreaInsets = useSafeAreaInsets();
 
-        this.hideModal = this.hideModal.bind(this);
-    }
+    /**
+     * Hides modal
+     * @param {Boolean} [callHideCallback=true] Should we call the onModalHide callback
+     */
+    const hideModal = (callHideCallback = true) => {
+        if (shouldSetModalVisibility) {
+            Modal.setModalVisibility(false);
+        }
+        if (callHideCallback) {
+            onModalHide();
+        }
+        Modal.onModalDidClose();
+    };
 
-    componentDidMount() {
-        if (!this.props.isVisible) {
+    useEffect(() => {
+        if (!isVisible) {
             return;
         }
 
         Modal.willAlertModalBecomeVisible(true);
 
         // To handle closing any modal already visible when this modal is mounted, i.e. PopoverReportActionContextMenu
-        Modal.setCloseModal(this.props.onClose);
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.isVisible === this.props.isVisible) {
-            return;
+        Modal.setCloseModal(onClose);
+
+        return () => {
+            // Only trigger onClose and setModalVisibility if the modal is unmounting while visible.
+            if (isVisible) {
+                hideModal(true);
+                Modal.willAlertModalBecomeVisible(false);
+            }
+
+            // To prevent closing any modal already unmounted when this modal still remains as visible state
+            Modal.setCloseModal(null);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- this effect only needs to run once on mount. It's an equivalent of componentDidMount and componentWillUnmount from the original class component
+    }, []);
+
+    useEffect(() => {
+        Modal.willAlertModalBecomeVisible(isVisible);
+        Modal.setCloseModal(isVisible ? onClose : null);
+    }, [isVisible, onClose]);
```

Comment:
Maybe this can be simplified with the other `useEffect` that calls the same functions

Reply:
Good idea, done!